src/sinks/datadog/traces/sink.rs

```diff
@@ -113,7 +120,33 @@ where
             })
             .into_driver(self.service);
 
-        sink.run().await
+        sink.run().await?;
+
+        // Signals the APM stats flushing thread to send the remaining cached payloads and exit.
+        {
+            // create a channel for the stats flushing thread to communicate back that it has flushed
+            // remaining stats. This is necessary so that we do not terminate the process while the
+            // stats flushing thread is trying to complete the HTTP request.
+            let (sender, mut receiver) = channel();
+
+            // Notify the stats thread task to flush remaining payloads and shutdown.
+            let _ = self.shutdown.send(sender);
+
+            // allow the stats flushing thread up to 50 seconds to send the remaining stats payloads
```

Comment:
```suggestion
            // Allow the stats flushing thread up to 50 seconds to send the remaining stats payloads
```

I also thought it was 60s 🤔

Reply:
Yeah I think this will be changing based on tobz's feedback.